@planetarium/tx/src/assets.ts

```diff
@@ -17,7 +17,7 @@ export function encodeCurrency(currency: Currency): Value {
     currency.minters === null ? null : encodeAddressSet(currency.minters);
   const serialized: RecordValue = {
     ticker: currency.ticker,
-    decimals: BigInt(currency.decimalPlaces),
+    decimalPlaces: new Uint8Array([currency.decimalPlaces]),
```

Comment:
From what I see in the Libplanet changes along with what I recall from the conversation with @riemannulus, the change of `Currency` serialization format was for compatibility with what `lib9c` had (which wasn't using `Currency.Serialize()` and had different format.) However, the currency hash is still used in different parts of the program, and it should still be the same, thus the need for `SerializeForHash()` for `GetHash()` (https://github.com/planetarium/libplanet/blob/main/Libplanet/Assets/Currency.cs#L747). Thus, I think this should be taken into account as well.